### ISSUE TITLE
Select the model based on testing accuracy？

### ISSUE THREAD

**Issue by PatrickZH** [floodsung/LearningToCompare_FSL#15]
Thank you for providing the code!
I have a concern about the model selection in your miniimagenet_train_few_shot.py. 
Line 260: It seems that the best training model is selected as the one with best testing accuracy (not validation accuracy) ?

**Comment by ehsanmok** [comment]
It's based on meta validation set: See [1](https://github.com/floodsung/LearningToCompare_FSL/blob/master/miniimagenet/task_generator.py#L29), [2](https://github.com/floodsung/LearningToCompare_FSL/blob/master/miniimagenet/miniimagenet_train_one_shot.py#L171), [3](https://github.com/floodsung/LearningToCompare_FSL/blob/master/miniimagenet/task_generator.py#L144)

**Comment by flexibility2** [comment]
@ehsanmok ,hi, however ,in the 
https://github.com/floodsung/LearningToCompare_FSL/blob/master/miniimagenet/miniimagenet_train_few_shot.py#L15
you just use the "task_generator_test", not the "task_generator"……

**Comment by ehsanmok** [comment]
Right! not a good code. It's the third mistake along with not using model.eval() and using the same normalization for omniglot and mini-imagenet!

**Comment by ehsanmok** [comment]
However, it's done correctly for one shot [here](https://github.com/floodsung/LearningToCompare_FSL/blob/master/miniimagenet/miniimagenet_train_one_shot.py#L15). Based on the copy pasting attitude of the code maybe it was changed at the time of training and when released the code wasn't carefully done!

**Comment by xyxxmb** [comment]
@ehsanmok Hello! I think that although the code uses "task_generator_test" instead of "task_generator" in miniimagenet_train_few_shot.py, it doesn't influence the result of model training because "metatest_folders" is only used for monitoring generalization performance. It doesn't participate in the process of model training.

**Comment by hyyuan123** [comment]
I would like to ask, is there a problem with the model selection based on omniglot?  Should be based on the accuracy of training to choose it

**Comment by hyyuan123** [comment]
When selecting a model, the test data is unknown, so the accuracy of the test cannot be used to select the model.

**Comment by Godxia** [comment]
> When selecting a model, the test data is unknown, so the accuracy of the test cannot be used to select the model.

I don't think so. FSL is not like a traditional training and testing procedure. When training, It can see a few labels of the test samples. It random chose many support set and query set in test data. So the accuracy of the test can be used to select the model.
